Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+@_spi(AsyncChannel)
+
+import NIOCore
+import NIOPosix
+
+public actor MemcachedConnection {
+    private let channel: NIOAsyncChannel<ByteBuffer, ByteBuffer>
+    private let bufferAllocator: ByteBufferAllocator
+    private let group: MultiThreadedEventLoopGroup
+    
+    public init(host: String, port: Int, group: MultiThreadedEventLoopGroup) async throws {
+            self.group = group
+            let bootstrap = ClientBootstrap(group: self.group)
+                .channelInitializer { channel in
+                    return channel.pipeline.addHandlers([MessageToByteHandler(MemcachedRequestEncoder()), ByteToMessageHandler(MemcachedResponseDecoder())])
+                }
+
+            let rawChannel = try await bootstrap.connect(host: host, port: port).get()
+            self.channel = try NIOAsyncChannel(synchronouslyWrapping: rawChannel, inboundType: ByteBuffer.self, outboundType: ByteBuffer.self)
```

Comment:
There is a newer method that avoids us wrapping the channel
```suggestion
            self.channel = try await bootstrap.connect(host: host, port: port)
```